### ISSUE TITLE
Apple Pay Error

### ISSUE THREAD

**Issue by SandyMental** [mollie/Shopware#328]
Never saw that before:

```
{
  "error": "Article with number:  not found!",
  "session": "hb08ki2bdtm57sogtsq7smkch6",
  "processors": {
    "uid": {
      "uid": "b2a8dd2"
    },
    "web": {
      "url": "/MollieApplePayDirect/startPayment",
      "ip": "127.0.0.*",
      "http_method": "POST",
      "server": "lalala.tld",
      "referrer": "https://lalala.tld/"
    },
    "introspection": {
      "file": null,
      "line": null,
      "class": "Shopware_Controllers_Frontend_MollieApplePayDirect",
      "function": "startPaymentAction"
    }
  }
}
```

**Comment by boxblinkracer** [comment]
Hi

huch....thanks for posting this.
Is it just once or every now and then?
do you have special products, like custom product things...where the product number might not be "standard"?

thanks

**Comment by SandyMental** [comment]
Nope, nothing like that. Just "normal" products with variants.

It happened once so far , but also Apple Pay is not very popular with our customers, so tracking and logging this will take some more time.
I thought I post this, just in case I'm not the only one.
Maybe you can close this for now until someone else jumps on this train :)


thanks

**Comment by dbMario** [comment]
Hi,
i have same problem.
I activate Apple Pay Direct and if i click the Apple Pay Button on Cart (Shopware 5), i can pay with Apple Pay, come back to Shop and get an error in Logfile. It looks like the variable productNumber is empty.
Did you find a solution for this problem?

**Comment by dbMario** [comment]
https://github.com/mollie/Shopware/wiki/Known-Issues#warning-mollie-is-paid-but-no-order-exists-in-shopware-for-transaction-xxxx-fixed-in-1810

This problem it is. Shopware Plugin version 2.0.0 is installed. Deactivate + activate + clear cache = problem still exists.

**Comment by boxblinkracer** [comment]
hi
i have a testing phase tomorrow anyway for shopware 5, and I'll check it out.
but still this is very weird, apple pay direct is a smoke test and tested before each release.
What Shopware version are you using?

another thing that could be is (i think there was something)...the apple pay direct webhooks come in so fast that this log entry exists, but it doesnt have any bad impact.
so do you also see something broken or "just" the log entry? because then its "just" a race condition with the log and webhooks (which is also bad, but not so bad :D )

**Comment by dbMario** [comment]
Hi,
i am using shopware 5.7.13.
i get an error 500 because the plugin throw an exception because of the missing productNumber.

**Comment by dbMario** [comment]
In method startPaymentAction in MollieApplePayDirect.php is in line 476 an exception if no productNumber is in param. But i don't know why this is required. I didn't start the process from a product page but from canvas cart. So there is no productNumber and in ApplePayButtonBuilder.php the variable only set if controller is detail.
I think this is an bug in the plugin.

**Comment by boxblinkracer** [comment]
perfect
its the Off-Canvas, I can reproduce it :)
thank you so much for this, I'll check it out

**Comment by dbMario** [comment]
I have seen your commit and copy the source to our testsystem. Problem solved 👍 
Nice job. We will wait for an update ;-)

**Comment by boxblinkracer** [comment]
:D ok that was fast
great to hear its working
about to test it and other features today, release should come on monday (no release before the weekend right :D )

**Comment by boxblinkracer** [comment]
released in 2.1.0
